components/ai_chat/core/common/mojom/ai_chat.mojom

```diff
@@ -409,7 +409,8 @@ interface AIChatUIHandler {
   ManagePremium();
   HandleVoiceRecognition(string conversation_uuid);
   ShowSoftKeyboard();
-  UploadImage() => (array<UploadedImage>? uploaded_images);
+  UploadImage(bool use_media_capture)
+      => (array<UploadedImage>? uploaded_images);
```

Comment:
Since `SetUseMediaCapture` is the android only interface, we should declare a new interface(ex. UploadImageWithMediaCapture) guarded by `[EnableIf=is_android]`.
And current `AIChatUIPageHandler::UploadImage` block would be `UploadImageInternal(bool use_media_capture...` called by both `UploadImage` and `UploadImageWithMediaCapture`

Reply:
per discussion on Slack,
with the complexity it will bring to the UI side, let's add a comment saying this arg is effective on android only